src/Pagination.ts

```diff
@@ -0,0 +1,103 @@
+import { v, w } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import PageLink, { PageLinkProperties } from './pagination/PageLink';
+
+import * as css from './styles/pagination.m.css';
+
+export const PaginationBase = ThemeableMixin(WidgetBase);
+
+export interface PaginationProperties extends ThemeableProperties {
+	page: number;
+	pages: number;
+	status: string;
+	onPageRequest?: (pageNumber: number) => void;
+}
+
+@theme(css)
+class Pagination extends PaginationBase<PaginationProperties> {
+	render (): DNode {
+		const {
+			onPageRequest,
+			page,
+			pages,
+			status
+		} = this.properties;
+		const isFirstPage = page === 1;
+		const isLastPage = page === pages;
+		const children: DNode[] = [
+			v('div', {
+				classes: this.classes(css.status)
+			}, [
+				status
+			])
+		];
+		const onclick = onPageRequest ? { onPageRequest } : {};
```

Comment:
I see functionally what is happening here, but it does feel a bit confusing.  Why is this named `onclick`?  It seems to either add `onPageRequest` or not depending on if it is in the properties.